third_party/pugixml-1.5/pugixml.cpp

```diff
@@ -7197,7 +7208,7 @@ PUGI__NS_BEGIN
 	PUGI__FN void convert_number_to_mantissa_exponent(double value, char* buffer, size_t buffer_size, char** out_mantissa, int* out_exponent)
 	{
 		// get a scientific notation value with IEEE DBL_DIG decimals
-		sprintf(buffer, "%.*e", DBL_DIG, value);
+		snprintf(buffer, buffer_size, "%.*e", DBL_DIG, value);
 		assert(strlen(buffer) < buffer_size);
```

Comment:
Fair enough - if this file is only used in tools and not at run-time, and the buffer overrun that it's guarding against won't happen, given the source of the XML data, I guess it's safe.  However, perhaps you could comment out the `assert` and indicate why, so that anyone who might need to look at this code years from now will understand.

Reply:
I changed my mind, and removed the `assert()` line.
I also removed the following meaningless line, `(void)!buffer_size;`.